account_reversal/models/account_move.py

```diff
@@ -0,0 +1,114 @@
+                'journal_id': journal_id,
+                'debit': credit,
+                'credit': debit,
+                'amount_currency': -amount_currency,
+            })
+        return lines
+
+    def _move_reverse_prepare(self, date=False, journal=False,
+                              move_prefix=False):
+        self.ensure_one()
+        journal = journal or self.journal_id
+        if journal.company_id != self.company_id:
+            raise UserError(
+                _("Wrong company Journal is '%s' but we have '%s'") % (
+                    journal.company_id.name, self.company_id.name))
+        ref = self.ref or move_prefix
+        if move_prefix and move_prefix != ref:
+            ref = ' '.join([move_prefix, ref])
+        date = date or self.date
+        move = self.copy_data()[0]
+        move.update({
+            'journal_id': journal.id,
+            'date': date,
+            'ref': ref,
+            'to_be_reversed': False,
+            'reversal_id': self.id,
+            'state': 'draft',
+        })
+        return move
+
+    @api.multi
+    def move_reverse_reconcile(self):
+        for move in self.filtered('reversal_id'):
+            rec = {}
+            lines = move.reversal_id.line_ids.filtered('account_id.reconcile')
+            for line in lines:
+                rec.setdefault((line.account_id, line.partner_id),
+                               self.env['account.move.line'])
+                rec[(line.account_id, line.partner_id)] += line
+            lines = move.line_ids.filtered('account_id.reconcile')
+            for line in lines:
+                rec[(line.account_id, line.partner_id)] += line
+            for lines in rec.itervalues():
+                lines.reconcile()
+        return True
+
+    @api.multi
+    def create_reversals(self, date=False, journal=False, move_prefix=False,
+                         line_prefix=False, post=False, reconcile=False):
+        """
+        Create the reversal of one or multiple moves
+
+        :param self: moves to reverse
+        :param date: when the reversal must be input
+                     (use original if empty)
+        :param journal: journal on which create the move
+                        (use original if empty)
+        :param move_prefix: prefix for the move's name
+        :param line_prefix: prefix for the move line's names
+        :param reconcile: reconcile lines (if account with reconcile = True)
+
+        :return: Returns a recordset of the created reversal moves
+        """
+        moves = self.env['account.move']
+        for orig in self:
+            data = orig._move_reverse_prepare(
+                date=date, journal=journal, move_prefix=move_prefix)
+            data['line_ids'] = orig._move_lines_reverse_prepare(
+                data.get('line_ids', []), date=date, journal=journal,
+                line_prefix=line_prefix)
+            moves |= self.create(data)
+        if moves and post:
```

Comment:
This option comes from the reversal wizard, we can default it to journal configuration